internal/service/advancedclustertpf/model_ClusterDescription20240805.go

```diff
@@ -0,0 +1,276 @@
+		PitEnabled:                                types.BoolPointerValue(input.PitEnabled),
+		RedactClientLogData:                       types.BoolPointerValue(input.RedactClientLogData),
+		ReplicaSetScalingStrategy:                 types.StringPointerValue(input.ReplicaSetScalingStrategy),
+		ReplicationSpecs:                          replicationSpecs,
+		RootCertType:                              types.StringPointerValue(input.RootCertType),
+		StateName:                                 types.StringPointerValue(input.StateName),
+		Tags:                                      tags,
+		TerminationProtectionEnabled:              types.BoolPointerValue(input.TerminationProtectionEnabled),
+		VersionReleaseSystem:                      types.StringPointerValue(input.VersionReleaseSystem),
+		Timeouts:                                  timeout,
+	}
+}
+
+func NewBiConnectorObjType(ctx context.Context, input *admin.BiConnector, diags *diag.Diagnostics) types.Object {
+	if input == nil {
+		return types.ObjectNull(BiConnectorObjType.AttrTypes)
+	}
+	tfModel := TFBiConnectorModel{
+		Enabled:        types.BoolPointerValue(input.Enabled),
+		ReadPreference: types.StringPointerValue(input.ReadPreference),
+	}
+	objType, diagsLocal := types.ObjectValueFrom(ctx, BiConnectorObjType.AttrTypes, tfModel)
+	diags.Append(diagsLocal...)
+	return objType
+}
+
+func NewConnectionStringsObjType(ctx context.Context, input *admin.ClusterConnectionStrings, diags *diag.Diagnostics) types.Object {
+	if input == nil {
+		return types.ObjectNull(ConnectionStringsObjType.AttrTypes)
+	}
+	privateEndpoint := NewPrivateEndpointObjType(ctx, input.PrivateEndpoint, diags)
+	tfModel := TFConnectionStringsModel{
+		AwsPrivateLink:    conversion.ToTFMapOfString(ctx, diags, input.AwsPrivateLink),
+		AwsPrivateLinkSrv: conversion.ToTFMapOfString(ctx, diags, input.AwsPrivateLinkSrv),
+		Private:           types.StringPointerValue(input.Private),
+		PrivateEndpoint:   privateEndpoint,
+		PrivateSrv:        types.StringPointerValue(input.PrivateSrv),
+		Standard:          types.StringPointerValue(input.Standard),
+		StandardSrv:       types.StringPointerValue(input.StandardSrv),
+	}
+	objType, diagsLocal := types.ObjectValueFrom(ctx, ConnectionStringsObjType.AttrTypes, tfModel)
+	diags.Append(diagsLocal...)
+	return objType
+}
+
+func NewLabelsObjType(ctx context.Context, input *[]admin.ComponentLabel, diags *diag.Diagnostics) types.Set {
+	if input == nil {
+		return types.SetNull(LabelsObjType)
+	}
+	tfModels := make([]TFLabelsModel, len(*input))
+	for i, item := range *input {
+		tfModels[i] = TFLabelsModel{
+			Key:   types.StringPointerValue(item.Key),
+			Value: types.StringPointerValue(item.Value),
+		}
+	}
+	setType, diagsLocal := types.SetValueFrom(ctx, LabelsObjType, tfModels)
+	diags.Append(diagsLocal...)
+	return setType
+}
+
+func NewMongoDBEmployeeAccessGrantObjType(ctx context.Context, input *admin.EmployeeAccessGrant, diags *diag.Diagnostics) types.Object {
```

Comment:
all NewXXXObjType could use a helper function similar to `NewObjectTF` in  https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2765/commits/a88e5642a517998ca01c093d7bf9ab550db8f735 to do the first and last part of the func, e.g:

```
func NewMongoDBEmployeeAccessGrantObjType(ctx context.Context, input *admin.EmployeeAccessGrant, diags *diag.Diagnostics) types.Object {
   return NewObjectTF(ctx, MongoDbEmployeeAccessGrantObjType, TFMongoDbEmployeeAccessGrantModel{
		ExpirationTime: types.StringValue(conversion.TimeToString(input.ExpirationTime)),
		GrantType:      types.StringValue(input.GrantType),
	}
)
}
```

Reply:
when i see boilerplate used in many funcs i tend to extract, but ok here